conclave-host/src/main/kotlin/com/r3/conclave/host/internal/attestation/AttestationServiceFactory.kt

```diff
@@ -10,13 +12,20 @@ object AttestationServiceFactory {
      * EPID and DCAP can only be used if the enclave is release or debug, and mock is only used if the enclave is
      * simulation or mock. In the later case any attestation parameters provided are ignored.
      */
-     fun getService(enclaveMode: EnclaveMode, attestationParameters: AttestationParameters?): AttestationService {
-        return when (enclaveMode) {
-            EnclaveMode.RELEASE -> getHardwareService(isRelease = true, enclaveMode, attestationParameters)
-            EnclaveMode.DEBUG -> getHardwareService(isRelease = false, enclaveMode, attestationParameters)
-            EnclaveMode.SIMULATION -> MockAttestationService(isSimulation = true)
-            EnclaveMode.MOCK -> MockAttestationService(isSimulation = false)
-        }
+    fun getService(
+        enclaveMode: EnclaveMode,
+        attestationParameters: AttestationParameters?,
+        enclaveHandle: EnclaveHandle
+    ): AttestationService {
+        // TODO: Fix attestation in Gramine: https://r3-cev.atlassian.net/browse/CON-1178
+        return if (enclaveHandle is GramineEnclaveHandle) MockAttestationService(isSimulation = true)
```

Comment:
I'd prefer this logic to be in `EnclaveHost.start`. It's not obvious to me that this `getService` will need to be affected by the Gramine attestation work. Or at least I'm not sure passing in the handle is the correct approach. Better to deal with this in that ticket.